src/MoBi.Presentation/Presenter/EditModuleConfigurationsPresenter.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MoBi.Core.Domain.Model;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Mappers;
+using MoBi.Presentation.Presenter.Simulation;
+using MoBi.Presentation.Views;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Presentation.Nodes;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Utility.Extensions;
+using ITreeNodeFactory = MoBi.Presentation.Nodes.ITreeNodeFactory;
+
+namespace MoBi.Presentation.Presenter
+{
+   public class EditModuleConfigurationsPresenter : AbstractSubPresenter<IEditModuleConfigurationsView, IEditModuleConfigurationsPresenter>, IEditModuleConfigurationsPresenter
+   {
+      private readonly ITreeNodeFactory _treeNodeFactory;
+      private readonly IMoBiContext _context;
+      private readonly IModuleConfigurationToModuleConfigurationDTOMapper _moduleConfigurationDTOMapper;
+      private List<ModuleConfigurationDTO> _moduleConfigurations;
+
+      public EditModuleConfigurationsPresenter(IEditModuleConfigurationsView view, ITreeNodeFactory treeNodeFactory, IMoBiContext context,
+         IModuleConfigurationToModuleConfigurationDTOMapper moduleConfigurationDTOMapper) : base(view)
+      {
+         _treeNodeFactory = treeNodeFactory;
+         _context = context;
+         _moduleConfigurationDTOMapper = moduleConfigurationDTOMapper;
+      }
+
+      public void Edit(SimulationConfiguration simulationConfiguration)
+      {
+         _moduleConfigurations = simulationConfiguration.ModuleConfigurations.Select(_moduleConfigurationDTOMapper.MapFrom).ToList();
+         
+         addUnusedModulesToSelectionView();
+         addUsedModuleConfigurationsToSelectedView(simulationConfiguration);
+      }
+
+      private void addUsedModuleConfigurationsToSelectedView(SimulationConfiguration simulationConfiguration)
+      {
+         simulationConfiguration.ModuleConfigurations.Each(addModuleConfigurationToSelectedView);
+      }
+
+      private void addUnusedModulesToSelectionView()
+      {
+         _context.CurrentProject.Modules.Where(module => !moduleInUse(module)).Each(addModuleToSelectionView);
+      }
+
+      private bool moduleInUse(Module module)
+      {
+         return _moduleConfigurations.Any(x => x.Uses(module));
+      }
+
+      private ModuleConfigurationDTO dtoFor(ModuleConfiguration moduleConfiguration)
+      {
+         return _moduleConfigurations.FirstOrDefault(x => x.Uses(moduleConfiguration));
+      }
+
+
+      private void addModuleToSelectionView(Module module)
+      {
+         _view.AddModuleNode(_treeNodeFactory.CreateFor(module));
```

Comment:
here it's called AddModuleNode. Above it should be AddxxExpressionNode and that would make sense to me